pinot-query-planner/src/main/java/org/apache/pinot/query/catalog/PinotCatalog.java

```diff
@@ -63,7 +63,11 @@ public PinotCatalog(TableCache tableCache) {
   @Override
   public Table getTable(String name) {
     String tableName = TableNameBuilder.extractRawTableName(name);
-    return new PinotTable(_tableCache.getSchema(tableName));
+    org.apache.pinot.spi.data.Schema schema = _tableCache.getSchema(tableName);
+    if (schema == null) {
+      throw new IllegalArgumentException("Could not find schema for table: " + tableName);
+    }
```

Comment:
nit: by the time it reaches here. it shouldn't be couldn't find the schema for a table but more of an internal error as PinotCatalog is inconsistent of the table it said it contains vs. the schema it can pull out from. 

but I don't know what's a better error message to indicate this info.

Reply:
I updated the error message, hopefully it's a bit better.